contract-message-predicate/src/main.sw

```diff
@@ -1,94 +1,170 @@
+fn verify_input_contract(index: u8) -> bool {
+    if let Input::Contract = input_type(index) {
+        true
+    } else {
+        false
+    }
 }
 
-/// Predicate verifying a message input is being spent according to the rules for a valid deposit
-fn main() -> bool {
-    ///////////////
-    // CONSTANTS //
-    ///////////////
+/// Verifies an input at the given index is a message input
+fn verify_input_message(index: u8) -> bool {
+    if let Input::Message = input_type(index) {
+        true
+    } else {
+        false
+    }
+}
 
-    // The minimum gas limit for the transaction not to revert out-of-gas.
-    const MIN_GAS = 42;
+/// Verifies an output at the given index is a contract output
+fn verify_output_contract(index: u8) -> bool {
+    if let Output::Contract = output_type(index) {
+        true
+    } else {
+        false
+    }
+}
 
-    // The hash of the script which must spend the input belonging to this predicate
-    // This ensures the coins can only be spent in a call to `TokenContract.finalizeDeposit()`
-    // Note: The script must be right-padded to the next full word before hashing, to match with `get_script_bytecode()`
-    const SPENDING_SCRIPT_HASH = 0x2d235589506d17993e0b7aca4407a5ac1c325efd9d704ff94696a8f7c012ab9d;
+/// Verifies an output at the given index is a change output
+fn verify_output_change(index: u8) -> bool {
+    if let Output::Change = output_type(index) {
+        true
+    } else {
+        false
+    }
+}
 
-    ////////////////
-    // CONDITIONS //
-    ////////////////
+/// Get the contract ID in the data of a message input
+fn input_message_contract_id(index: u64) -> ContractId {
+    // Length should be at least 32 bytes for the contract ID
+    let msg_data_len = input_message_data_length(index);
+    assert(msg_data_len >= 32);
 
-    // Verify script bytecode hash matches
-    let script_bytcode: [u64;
-    32] = tx_script_bytecode(); // Note : Make sure length is script bytecode length rounded up to next word
-    assert(sha256(script_bytcode) == SPENDING_SCRIPT_HASH);
+    // Parse the contract ID from the message data
+    let contract_id: b256 = input_message_data(index, 0);
+    ~ContractId::from(contract_id)
+}
 
-    // Verify gas limit is high enough
-    //TODO: does gas limit include InputMessage amount? might need to just check avail balance - InputMessage amount
-    assert(tx_gas_limit() >= MIN_GAS);
+/// Verifies the input at the given index meets expectations (returns amount of base asset coins)
+fn verify_other_input(index: u8, num_inputs: u8) -> u64 {
+    let mut num_coins: u64 = 0;
+    if(index < num_inputs) {
+        match input_type(index) {
+            Input::Coin => {
+                // Coin inputs must be of the base asset
+                assert(input_coin_asset_id(index) == BASE_ASSET_ID);
+                num_coins = input_coin_amount(index);
+            },
+            Input::Contract => {
+                // Additional contract inputs are allowed
```

Comment:
Should the `Contract` branch have code? Similar to the function below. There are comments but no code

Reply:
No, it just doesn't `assert(false)`, so it needs its own segment outside of the `_` catch all